docs/SANITY_MIGRATION_PLAN.md

```diff
@@ -0,0 +1,765 @@
+**Estimated Effort:** Medium
+
+**Context:**
+All migration scripts need common functionality for image uploads, HTML conversion, validation, and logging. Creating shared utilities prevents code duplication and ensures consistency.
+
+**Requirements:**
+1. Create `bin/migrations/shared/` folder with:
+   - `imageUploader.ts` - Upload images to Sanity, return asset reference
+   - `richTextConverter.ts` - Convert HTML to Portable Text
+   - `validation.ts` - Validate records, return errors
+   - `logger.ts` - Consistent logging with timestamps
+   - `upsert.ts` - Upsert logic using `_legacyId` for matching
+2. Use existing `bin/migrations/sanity.ts` patterns as reference
+
+**Files to Create:**
+- `bin/migrations/shared/imageUploader.ts`
+- `bin/migrations/shared/richTextConverter.ts`
+- `bin/migrations/shared/validation.ts`
+- `bin/migrations/shared/logger.ts`
+- `bin/migrations/shared/upsert.ts`
+- `bin/migrations/shared/index.ts` (barrel export)
+
+**Acceptance Criteria:**
+- [ ] All utilities compile without TypeScript errors
+- [ ] Image uploader handles external URLs (e.g., Imgur)
+- [ ] Rich text converter produces valid Portable Text
+- [ ] Upsert finds existing records by `_legacyId`
+
+---
+
+### Task 3: Create Ad Schema and Migration
+
+**Status:** 🔲 Not Started  
+**Depends On:** Task 2 (Shared Utilities)  
+**Estimated Effort:** Small
+
+**Context:**
+The `ads` table stores sponsor/advertisement data with images and date ranges. Simple model, good for establishing migration patterns.
+
+**MySQL Schema:**
+```
+ads: id, name, start_date, end_date, pic_url, web_url, priority, deleted
+```
+
+**Requirements:**
+1. Create `studio/schemaTypes/ad.ts`:
+   - `name` (string, required)
+   - `startDate` (date)
+   - `endDate` (date)
+   - `image` (image)
+   - `url` (url)
+   - `priority` (number)
+   - `_legacyId`, `_migratedAt`
+2. Create `bin/migrations/importAds.ts`
+3. Add npm script `import:ads`
+4. Generate migration report
+
+**Files to Modify:**
+- `studio/schemaTypes/ad.ts` (create)
+- `studio/schemaTypes/index.ts` (update)
+- `bin/migrations/importAds.ts` (create)
+- `package.json` (add script)
+
+**Acceptance Criteria:**
+- [ ] Schema works in Sanity Studio
+- [ ] Migration script runs without errors
+- [ ] Images migrated to Sanity assets
+- [ ] Soft-deleted records (deleted='y') not migrated
+- [ ] Migration report generated
+
+---
+
+### Task 4: Create Concert Schema and Migration
+
+**Status:** 🔲 Not Started  
+**Depends On:** Task 1 (Artist Schema), Task 2 (Shared Utilities)  
+**Estimated Effort:** Medium
+
+**Context:**
+Concerts reference artists and include venue/ticket information.
+
+**MySQL Schema:**
+```
+concerts: id, date, artist, band_pic_url, band_url, venue, ticketinfo, ticketurl, featured, deleted
+```
+
+**Requirements:**
+1. Create `studio/schemaTypes/concert.ts`:
+   - `date` (date, required)
+   - `artist` (reference to `artist`)
+   - `artistName` (string - for display if no artist ref)
```

Comment:
There should always be an artist record, even if the user has to make one on the fly.